lldb/source/Plugins/Language/Swift/FoundationValueTypes.cpp

```diff
@@ -96,13 +96,52 @@ bool lldb_private::formatters::swift::URL_SummaryProvider(
     return false;
 
   std::string summary;
-  if (!underlying_url_sp->GetSummaryAsCString(summary, options))
+  auto try_format = [&]() {
+    if (underlying_url_sp->GetSummaryAsCString(summary, options))
+      return true;
+    auto dynval_sp =
+        underlying_url_sp->GetDynamicValue(lldb::eDynamicCanRunTarget);
```

Comment:
should the default be hardcoded? Is there not a value provided to the formatter somewhere?

Reply:
which cases does the static value fail, but the dynamic value succeeds?

Reply:
Older versions of Foundation have a storage member containing the ObjC implementation, newer ones may have an existential that is a _SwiftURL